qa/rolling-upgrade/src/test/java/org/opensearch/knn/bwc/LuceneFilteringIT.java

```diff
@@ -0,0 +1,86 @@
+ */
+public class LuceneFilteringIT extends AbstractRollingUpgradeTestCase {
+    private static final String TEST_FIELD = "test-field";
+    private static final int DIMENSIONS = 50;
+    private static final int K = 10;
+    private static final int NUM_DOCS = 100;
+    private static final TermQueryBuilder TERM_QUERY = QueryBuilders.termQuery("_id", "100");
+
+    public void testLuceneFiltering() throws Exception {
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+        float[] queryVector = TestUtils.getQueryVectors(1, DIMENSIONS, NUM_DOCS, true)[0];
+        switch (getClusterType()) {
+            case OLD:
+                createKnnIndex(testIndex, getKNNDefaultIndexSettings(), createKnnIndexMappingWithLuceneField(TEST_FIELD, DIMENSIONS));
+                bulkAddKnnDocs(testIndex, TEST_FIELD, TestUtils.getIndexVectors(NUM_DOCS, DIMENSIONS, true), NUM_DOCS);
+                validateSearchKNNIndexFailed(testIndex, new KNNQueryBuilder(TEST_FIELD, queryVector, K, TERM_QUERY), K);
+                break;
+            case MIXED:
+                validateSearchKNNIndexFailed(testIndex, new KNNQueryBuilder(TEST_FIELD, queryVector, K, TERM_QUERY), K);
+                break;
+            case UPGRADED:
+                searchKNNIndex(testIndex, new KNNQueryBuilder(TEST_FIELD, queryVector, K, TERM_QUERY), K);
+                deleteKNNIndex(testIndex);
+                break;
+        }
+    }
+
+    protected String createKnnIndexMappingWithLuceneField(final String fieldName, int dimension) throws IOException {
+        return Strings.toString(
+            XContentFactory.jsonBuilder()
+                .startObject()
+                .startObject("properties")
+                .startObject(fieldName)
+                .field("type", "knn_vector")
+                .field("dimension", Integer.toString(dimension))
+                .startObject("method")
+                .field("name", "hnsw")
+                .field("engine", "lucene")
+                .field("space_type", "l2")
+                .endObject()
+                .endObject()
+                .endObject()
+                .endObject()
+        );
+    }
+
+    private void validateSearchKNNIndexFailed(String index, KNNQueryBuilder knnQueryBuilder, int resultSize) throws IOException {
+        XContentBuilder builder = XContentFactory.jsonBuilder().startObject().startObject("query");
+        knnQueryBuilder.doXContent(builder, ToXContent.EMPTY_PARAMS);
+        builder.endObject().endObject();
+
+        Request request = new Request("POST", "/" + index + "/_search");
+
+        request.addParameter("size", Integer.toString(resultSize));
+        request.addParameter("explain", Boolean.toString(true));
+        request.addParameter("search_type", "query_then_fetch");
+        request.setJsonEntity(Strings.toString(builder));
+
+        expectThrows(ResponseException.class, () -> client().performRequest(request));
```

Comment:
What if this fails with another unrelated exception? i.e. are we sure this will work in a valid case?

Reply:
It's possible to get same exception type for a different cause. Let me add one more assert for the error message, I think that should be specific to our case